platform/linux-generic/odp_classification.c

```diff
@@ -1782,3 +1782,95 @@ uint64_t odp_pmr_to_u64(odp_pmr_t hdl)
 {
 	return _odp_pri(hdl);
 }
+
+static
+void print_cos_ident(struct cos_s *cos)
+{
+	ODP_PRINT("%s(%d)\n", cos->name, (int)cos->index);
+}
+
+static
+void print_queue_ident(odp_queue_t q)
+{
+	odp_queue_info_t info;
+
+	if (odp_queue_info(q, &info))
+		ODP_PRINT("%" PRIx64, odp_queue_to_u64(q));
+	else
+		ODP_PRINT("%s", info.name);
+
+	ODP_PRINT("\n");
+}
+
+static
+void print_hex(const void *vp, int len)
+{
+	const uint8_t *p = vp;
+
+	for (int i = 0; i < len; i++)
+		ODP_PRINT("%02x", (int)*p++ & 0xff);
+}
+
+static
+void cls_print_cos(struct cos_s *cos)
+{
+	print_cos_ident(cos);
+	ODP_PRINT("    queue: ");
+	print_queue_ident(cos->queue);
+
+	uint32_t num_rule = odp_atomic_load_u32(&cos->num_rule);
+	bool first = true;
+
+	for (uint32_t j = 0; j < num_rule; j++) {
+		struct pmr_s *pmr = &cos->pmr[j]->s;
+
+		LOCK(&pmr->lock);
+		for (uint32_t k = 0; k < pmr->num_pmr; k++) {
+			if (first)
+				ODP_PRINT("    rules: ");
+			else
+				ODP_PRINT("           ");
+
+			first = false;
+
+			pmr_term_value_t *v = &pmr->pmr_term_value[k];
+
+			ODP_PRINT("%s: ", format_pmr_name(v->term));
+
+			if (v->term == ODP_PMR_CUSTOM_FRAME ||
+			    v->term == ODP_PMR_CUSTOM_L3)
+				ODP_PRINT("offset:%" PRIu32 " ", v->offset);
+
+			if (v->range_term) {
+				ODP_PRINT("<range>");
+			} else {
+				print_hex(v->match.value_u8, v->val_sz);
+				ODP_PRINT(" ");
+				print_hex(v->match.mask_u8, v->val_sz);
+			}
+
+			ODP_PRINT(" -> ");
+
+			if (pmr->mark)
+				ODP_PRINT("mark:%" PRIu16 " ", pmr->mark);
+
+			print_cos_ident(&cos->linked_cos[j]->s);
+		}
+		UNLOCK(&pmr->lock);
+	}
+}
+
+void odp_cls_print_all(void)
+{
+	ODP_PRINT("Classifier info\n"
```

Comment:
Info prints usually start with a new line. That way those are easier to spot in the middle of a log, especially if the previous print didn't end with a new line.